deepcell_spots/applications/polaris.py

```diff
@@ -23,227 +23,142 @@
+        cells of the input.
+
         Input images are required to have 4 dimensions
-        ``[batch, x, y, channel]``.
+        ``[batch, x, y, channel]``. Channel dimension should be 2.
+
         Additional empty dimensions can be added using ``np.expand_dims``.
+
         Args:
             image (numpy.array): Input image with shape
                 ``[batch, x, y, channel]``.
-            batch_size (int): Number of images to predict on per batch.
             image_mpp (float): Microns per pixel for ``image``.
-            pad_mode (str): The padding mode, one of "constant" or "reflect".
-            preprocess_kwargs (dict): Keyword arguments to pass to the
-                pre-processing function.
-            postprocess_kwargs (dict): Keyword arguments to pass to the
-                post-processing function.
+            cytoplasm_channel (int): Value should be 0 or 1 depending on the channel
+                containing the images for cell segmentation. Defaults to 0.
+            spots_channel (int): Value should be 0 or 1 depending on the channel
+                containing the images for spot detection. Defaults to 1.
             threshold (float): Probability threshold for a pixel to be
-                                considered as a spot.
+                considered as a spot.
+            clip (bool): Determines if pixel values will be clipped by percentile.
+                Defaults to false.
         Raises:
             ValueError: Input data must match required rank of the application,
-                calculated as one dimension more (batch dimension) than
-                expected by the model.
+                calculated as one dimension more (batch dimension) than expected
+                by the model.
             ValueError: Input data must match required number of channels.
+            ValueError: Threshold value must be between 0 and 1.
+            ValueError: Segmentation application must be instantiated if segmentation
+                image is defined.
+
         Returns:
-            numpy.array: Coordinate locations of detected spots.
+            list: List of dictionaries, length equal to batch dimension.
         """
 
-        if threshold < 0 or threshold > 1:
-            raise ValueError("""Enter a probability threshold value between
-                                0 and 1.""")
-
-        if preprocess_kwargs is None:
-            preprocess_kwargs = {}
-
-        if postprocess_kwargs is None:
-            postprocess_kwargs = {
-                'threshold': threshold,
-                'min_distance': 1}
-
-        return self._predict_segmentation(
-            image,
-            batch_size=batch_size,
-            image_mpp=image_mpp,
-            pad_mode=pad_mode,
-            preprocess_kwargs=preprocess_kwargs,
-            postprocess_kwargs=postprocess_kwargs)
+        if spots_threshold < 0 or spots_threshold > 1:
```

Comment:
Is it correct that the threshold should not be inclusive of 0 or 1?

Reply:
As written, the threshold is inclusive, because it only raises an error if the value is less than zero or greater than 1. Having the threshold value be inclusive of 0 makes more sense than 1, but I think both should be valid entries.